giraffe/src/utils/fromFlux.ts

```diff
@@ -189,6 +189,162 @@ export const fromFlux = (fluxCSV: string): FromFluxResult => {
+
+      const tableTexts = []
+      const annotationTexts = []
+
+      splittedChunk.forEach(line => {
+        if (line.startsWith('#')) {
+          annotationTexts.push(line)
+        } else {
+          tableTexts.push(line)
+        }
+      })
+
+      tableText = tableTexts.join('\n').trim()
+
+      assert(
+        !!tableText,
+        'could not find annotation lines in Flux response; are `annotations` enabled in the Flux query `dialect` option?'
+      )
+
+      // TODO(ariel): csvParse is a slow operation
+      tableData = csvParse(tableText)
+
+      annotationText = annotationTexts.join('\n').trim()
+
+      assert(
+        !!annotationText,
+        'could not find annotation lines in Flux response; are `annotations` enabled in the Flux query `dialect` option?'
+      )
+      const annotationData = parseAnnotations(annotationText, tableData.columns)
+
+      for (const columnName of tableData.columns.slice(1)) {
+        columnType =
+          TO_COLUMN_TYPE[annotationData.datatypeByColumnName[columnName]]
+
+        assert(
+          !!columnType,
+          `encountered unknown Flux column type ${annotationData.datatypeByColumnName[columnName]}`
+        )
+
+        columnKey = `${columnName} (${columnType})`
+
+        if (!columns[columnKey]) {
+          columns[columnKey] = {
+            name: columnName,
+            type: columnType,
+            fluxDataType: annotationData.datatypeByColumnName[columnName],
+            data: [],
+          } as Column
+        }
+
+        columnDefault = annotationData.defaultByColumnName[columnName]
+
+        for (let i = 0; i < tableData.length; i++) {
+          if (columnName === RESULT) {
+            if (columnDefault.length) {
+              resultColumnNames.add(columnDefault)
+            } else if (tableData[i][columnName].length) {
+              resultColumnNames.add(tableData[i][columnName])
+            }
+          }
+          const value = tableData[i][columnName] || columnDefault
+          let result = null
+
+          if (value === undefined) {
+            result = undefined
+          }
+
+          if (value === 'null') {
+            result = null
+          }
+
+          if (value === 'NaN') {
+            result = NaN
+          }
+
+          if (columnType === 'boolean' && value === 'true') {
+            result = true
+          }
+
+          if (columnType === 'boolean' && value === 'false') {
+            result = false
+          }
+
+          if (columnType === 'string') {
+            result = value
+          }
+
+          if (columnType === 'time') {
+            result = Date.parse(value.trim())
+          }
+
+          if (columnType === 'number') {
+            if (value === '') {
+              result = null
+            }
+            const parsedValue = Number(value)
+            result = parsedValue === parsedValue ? parsedValue : value
+          }
+
+          columns[columnKey].data[tableLength + i] = result
```

Comment:
we sacrifice readability for memory allocation improvements by removing the `parseValue` usage in the function and simply inlining the operation